src/utils.zig

```diff
@@ -408,6 +459,15 @@ pub fn FixedSizeCircularBuffer(T: type, capacity: comptime_int) type { // MARK:
 			if(i >= self.len) return null;
 			return self.mem[(self.startIndex + i) & mask];
 		}
+
+		pub fn isEmpty(self: Self) bool {
+			return self.len == 0;
+		}
+
+		pub fn isFull(self: Self) bool {
```

Comment:
This is an internal helper function and should be private.
I can't think of any use case for using this externally.

Reply:
Also maybe the same thing applies as for isEmpty:
Why not just keep this in the places where it's called?
E.g. functions like `insertSlice` also explicitly check against the capacity.